### PR TITLE
fix: error when receiving an AppWrapper with 0 or 1 ordered instances

### PR DESCRIPTION
**Pull request by VanillaSpoon**
This pr resolves  https://github.com/project-codeflare/instascale/issues/69  https://github.com/project-codeflare/instascale/issues/76 and #35

### CODE REVIEW

**Review by asm582 — APPROVED** [review]
LGTM 🎉

**Comment by VanillaSpoon** [comment]
Thanks for the review @asm582 :)

File: controllers/appWrapper_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import (
+	"testing"
+
+	"github.com/onsi/gomega"
+	arbv1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestDiscoverInstanceTypes(t *testing.T) {
```
Comment (dimakis) [inline controllers/appWrapper_controller_test.go:11]: fantastic to see tests being added 👍

**Review by dimakis — COMMENTED** [review]
I've not validated it but the code looks good and the tests pass. 
I'll try validate it next week, great work @VanillaSpoon

**Review by KPostOffice — COMMENTED** [review]
Looks good! I have a question for @asm582 or @Maxusmusti though.

Does instascale assume only a single pod per new node created?

**Comment by VanillaSpoon** [comment]
Performed some additional checks to see if it resoved another issue, and had this moved to wip. 

But I'm happy with how the additional checks turned out, the other issue has been added to the description.

@dimakis if you'd like to have a look

**Review by asm582 — APPROVED** [review]
LGTM 🎸

**Comment by asm582** [comment]
> Looks good! I have a question for @asm582 or @Maxusmusti though.
> 
> Does instascale assume only a single pod per new node created?

Yup, hence the label orderedInstances

**Pull request merged**